### ISSUE TITLE
Can't obtain 'npb-benchmark-suite' resources as `x86-npb` has missing v24

### ISSUE THREAD

**Issue by wkjung** [gem5/gem5#1469]
**Describe the bug**

I cannot obtain the resource 'npb-benchmark-suite' from v24, especially because of `x86-npb`.

```py
#a.py
from gem5.resources.resource import obtain_resource
obtain_resource("npb-benchmark-suite")
```

Terminal output:
```
$ build/ALL/gem5.opt -v a.py 
gem5 Simulator System.  https://www.gem5.org
gem5 is copyrighted software; use the --copyright option for details.

gem5 version 24.0.0.1
gem5 compiled Aug 16 2024 13:16:46
gem5 started Aug 16 2024 13:21:11
gem5 executing on **, pid 16092
command line: build/ALL/gem5.opt -v a.py

info: Using default config
Exception: Resource with ID 'x86-npb' not found.

At:
  src/python/gem5/resources/client.py(280): _get_resource_json_obj_from_client
  src/python/gem5/resources/client.py(198): get_multiple_resource_json_obj
  src/python/gem5/resources/resource.py(1128): _get_workload
  src/python/gem5/resources/resource.py(1073): _get_suite
  src/python/gem5/resources/resource.py(998): obtain_resource
  a.py(2): <module>
  src/python/m5/main.py(669): main
```

Similarly `obtain_resource("x86-npb")` doesn't work with the same message.

But when I manually hack the `atlasclient.py` from 
```
                condition["gem5_versions"] = {
                    "$regex": f"^{resource.get_gem5_version()}", # (which chooses v24 here)
                    "$options": "i",
                }
```
to

```
                condition["gem5_versions"] = {
                    "$regex": f"^23.1",
                    "$options": "i",
                }
```

it worked. 

[`npb-benchmark-suite` is said to be working w/ v24](https://resources.gem5.org/resources/npb-benchmark-suite/versions?database=gem5-resources&version=1.0.0), so `x86-npb` should also support `v24`, I think? 

(btw, out of curiosity, then how the current CI on benchmark suites succeed in fetching the resource?)

**Affects version**
v24.0.0.1 (b1a44b89c7)

**gem5 Modifications**
No

**To Reproduce**
```shell
scons build/ALL/gem5.opt 
build/ALL/gem5.opt -v a.py 
```

**Host Operating System**
macOS

**Host ISA**
arm

**Compiler used**
Apple clang 15

**Comment by Harshil2107** [comment]
This error is due to a change that was pushed to resources in gem5 that dont get the resource if it is not compatible with the gem5_version. 

I will push a PR that updates this part of the code and throws a warning instead, but still returns the resources.

**Comment by BobbyRBruce** [comment]
Fixed by #1475
